### ISSUE TITLE
Resolve error - CE crashes after rename Excel.Sheet2 to a different name

### ISSUE THREAD

**Issue by PeterMTaylor** [rubberduck-vba/Rubberduck#1816]
This is really, really good. I loooove the performance of RD at the moment much better.
My VS2015 is synced to https://github.com/rubberduck-vba/Rubberduck/commit/a594e0435d6c878850214eafaf83fdcdf77a7ac1
CE suggests renaming "class" sheet2 into another name which then threw an exception error because I did not have Sheet2 in my workbook before the error occured, it was called "walgreen". CE did not mentioned I have Sheet2 as Walgreen, should this have been suggested to me to prevent renaming (which is deleting and/or creating a sheet with a different name?)
```
Rubberduck.App : 2016-06-14 21:14:20.9178;DEBUG;Rubberduck.App;Component 'Sheet2' was renamed to 'NextData'.;

Rubberduck.UI.Inspections.InspectionResultsViewModel : 2016-06-14 21:14:20.9647;DEBUG;Rubberduck.UI.Inspections.InspectionResultsViewModel;InspectionResultsViewModel handles StateChanged...;

Exception thrown: 'System.NotSupportedException' in Rubberduck.dll
Exception thrown: 'System.Runtime.InteropServices.COMException' in Rubberduck.dll
Additional information: Unknown name. (Exception from HRESULT: 0x80020006 (DISP_E_UNKNOWNNAME))
```
![capture sheet2 crashed resolving](https://cloud.githubusercontent.com/assets/2254567/16041262/b97b6f06-3277-11e6-9579-2becc8049ade.PNG)
![capture sheet2 error call stack](https://cloud.githubusercontent.com/assets/2254567/16041261/b97add66-3277-11e6-97f5-069ae3eadcf0.PNG)
![capture sheet2 error](https://cloud.githubusercontent.com/assets/2254567/16041263/b97c6b54-3277-11e6-897c-3db99c49a461.PNG)

**Comment by Hosch250** [comment]
No repro here on the latest commit.

**Comment by Hosch250** [comment]
No, renaming a sheet *should* just be changing the hidden "_CodeName" property (is that the right name?).  No deleting/creating involved.

**Comment by Hosch250** [comment]
After I do tonight's release, could you check this again?  I fixed a bug with the code explorer folders recently that looks like it could have been the cause of this based on the stack trace.

**Comment by PeterMTaylor** [comment]
ok @Hosch250 , have installed the latest fix 2.0.3. A couple of UX observations to get out of the way.
On the plus side, I like now that we are waiting for "Resolving Declarations" etc as opposed to "Resolving..." which doesn't mean anything. On the neg side, took a lot longer for CE to display the output 2.0.2 vs 2.0.3 was waiting watching the purple spinning wheel well over 10 mins. (enough time for me to wash my dishes from afternoon tea) which I received more Code Quality Issues. Could be false positives, don't know at this stage. 

![capture 2 0 3 ce sheet 2 test](https://cloud.githubusercontent.com/assets/2254567/16354988/c00e2188-3aec-11e6-9a4f-94538960667a.PNG)

I had to find the part that asks me to rename sheet two and so I did.
![capture 2 0 3 ce sheet 2 test 3](https://cloud.githubusercontent.com/assets/2254567/16355202/03435490-3af3-11e6-98c0-488e1955bc20.PNG)

Well done! No crashes when I did the rename and had to wait again about 5-10 mins for the refresh of the CE display to finish and see what happened, the Code Inspection for Sheet 2 disappeared where the blue inkmarks shows.

![capture 2 0 3 ce sheet 2 test 4](https://cloud.githubusercontent.com/assets/2254567/16355251/cf5e1c6c-3af4-11e6-845e-47bcf4610bf1.PNG)
